agent/consul/rpc.go

```diff
@@ -434,5 +434,31 @@ func (s *Server) setQueryMeta(m *structs.QueryMeta) {
 func (s *Server) consistentRead() error {
 	defer metrics.MeasureSince([]string{"consul", "rpc", "consistentRead"}, time.Now())
 	future := s.raft.VerifyLeader()
-	return future.Error()
+	err := future.Error()
+	if err != nil {
+		return err //fail fast if leader verification fails
+	}
+	// poll consistent read readiness, wait for up to RPCHoldTimeout milliseconds
+	if s.isReadyForConsistentReads() {
+		return nil
+	}
+	jitter := lib.RandomStagger(s.config.RPCHoldTimeout / jitterFraction)
+	deadline := time.Now().Add(jitter)
```

Comment:
I think this'll just wait one `jitterFraction` of the overall `RPCHoldTimeout`. If you set deadline to `time.Now().Add(s.config.RPCHoldTimeout)` that should work. The jitter basically is the poll interval.